command/agent/csi_endpoint.go

```diff
@@ -409,7 +409,7 @@ func parseCSISecrets(req *http.Request) structs.CSISecrets {
 	secrets := map[string]string{}
 	secretkvs := strings.Split(secretsHeader, ",")
 	for _, secretkv := range secretkvs {
-		kv := strings.Split(secretkv, "=")
+		kv := strings.SplitN(secretkv, "=", 2)
 		if len(kv) == 2 {
 			secrets[kv[0]] = kv[1]
 		}
```

Comment:
What about strings.Cut? The docs for SplitN suggest it for the case of splitting at the first separator. Might be a little easier to read?

```suggestion
		if key, value, found := strings.Cut(secretkv, "="); found {
			secrets[key] = value
		}
```

Reply:
Good catch, will fix

Reply:
TIL, nice!